File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:

    bool ispalindrome(string str ,int i ,int j ){

        while(i<j){
            if(str[i] != str[j]){
                return false;
            }
            i++;
            j--;
        }
        return true;

    }
    bool validPalindrome(string s) {
        int n = s.length();

        int i=0;
        int j=n-1;
        

        while(i < j){
            if(s[i] == s[j]){
                i++;
                j--;
            }else{
                return ispalindrome(s, i+1, j) || ispalindrome(s, i, j-1);                               
            }
             
        }     
        return true;
    }
};

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution {

    public boolean ispalindrome(int i, int j, String s)
    {
        while(i <j)
        {
            if(s.charAt(i) != s.charAt(j))
            {
                return false;
            }

            i+=1;
            j-=1;
        }
        return true;
    }
    public boolean validPalindrome(String s) {
        int i = 0;
        int j = s.length() - 1;

        while(i < j)
        {
            char left = s.charAt(i);
            char right = s.charAt(j);

            if(left != right)
            {
                return ispalindrome(i+1 , j ,s) || ispalindrome(i,j-1,s);
            }
            else
            {
                i+=1;
                j-=1;
            }
        }

        return true;
    }
}